File: boardinghouse_backend/app/Models/UtilBoardingHouse.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class UtilBoardingHouse extends Model
{
    use HasFactory;

    protected $fillable = [
        'util_id',
        'boarding_house_id',
    ];

    /**
     * Mối quan hệ thuộc về với model Util.
     *
     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
     */
    public function util()
    {
        return $this->belongsTo(Util::class);
    }

    /**
     * Mối quan hệ thuộc về với model BoardingHouse.
     *
     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
     */
    public function boardingHouse()
    {
        return $this->belongsTo(BoardingHouse::class);
    }
}

File: boardinghouse_app/lib/models/util_boarding_house.dart
class UtilBoardingHouse {
  int? id;
  int? utilId;
  int? boardingHouseId;

  UtilBoardingHouse({
    this.id,
    this.utilId,
    this.boardingHouseId,
  });

  factory UtilBoardingHouse.fromJson(Map<String, dynamic> json) {
    return UtilBoardingHouse(
      id: json['id'],
      utilId: json['util_id'],
      boardingHouseId: json['boarding_house_id'],
    );
  }
}
